### PR TITLE
perf(srt): avoid O(bs^2) extend prefix sums in prefill Triton hot path

### PR DESCRIPTION
**Pull request by guzekai01**
## Motivation
On the prefill hot path, compute_position_triton and write_req_to_token_pool_triton previously performed per-program prefix accumulation over extend_seq_lens / extend_lens inside the kernel, which scales as O(bs²) and becomes costly at large batch sizes.

This PR precomputes exclusive start offsets as extend_start_loc (same length as extend_lens) on the scheduling side and reuses that tensor across ScheduleBatch → ModelWorkerBatch → ForwardBatch and in write_cache_indices, so Triton kernels no longer redo equivalent prefix work.

## Modifications
- **python/sglang/srt/utils/common.py**: Add compute_start_loc_from_lens(seq_lens) to unify “per-sequence length tensor → exclusive start indices.”
- **python/sglang/srt/managers/schedule_batch.py**: Add and populate extend_start_loc on ScheduleBatch / ModelWorkerBatch; maintain it in prepare_for_extend and mix_with_running; pass it through ModelWorkerBatch.
- **python/sglang/srt/model_executor/forward_batch_info.py**: Extend compute_position / compute_position_triton / compute_position_torch with optional extend_start_loc; Triton kernel loads precomputed offsets instead of inner accumulation loops; ForwardBatch.init_new passes batch.extend_start_loc into compute_position.
- **python/sglang/srt/mem_cache/common.py**: write_req_to_token_pool_triton and write_cache_indices take extend_start_loc; non-Triton path uses compute_start_loc_from_lens once for per-row start offsets.

## Accuracy Tests
- server: kimi-k2.5  tp16
``` bash
#python -m sglang.test.few_shot_gsm8k   --host 127.0.0.1   --port 30000   --num-questions 200   --num-shots 5                                                                                    
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 200/200 [00:21<00:00,  9.38it/s]
Accuracy: 0.965
Invalid: 0.000
Latency: 21.496 s
Output throughput: 905.849 token/s
```

## Checklist

- [x] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [x] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [x] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [x] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [x] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/model_executor/forward_batch_info.py [hunk 1]
```diff
@@ -1088,33 +1089,42 @@ def compute_position(
...
     """Compute positions. It is a fused version of `compute_position_torch`."""
     batch_size = extend_seq_lens.shape[0]
     has_prefix = extend_prefix_lens.shape[0] == batch_size
 
+    if (
+        extend_start_loc is None
+        or extend_start_loc.shape[0] != batch_size
+        or extend_start_loc.device != extend_seq_lens.device
+    ):
+        extend_start_loc = compute_start_loc_from_lens(extend_seq_lens)
```
[the remark this answers is not in the document]
Comment (guzekai01) [reply python/sglang/srt/model_executor/forward_batch_info.py]: Done

File: python/sglang/srt/managers/schedule_batch.py [hunk 2]
```diff
@@ -1519,6 +1524,11 @@ def prepare_for_extend(self):
         self.seq_lens = seq_lens_tensor
         self.seq_lens_cpu = seq_lens_cpu
         self.extend_num_tokens = extend_num_tokens
+        self.extend_start_loc = compute_start_loc_from_lens(
+            torch.tensor(extend_lens, dtype=torch.int32, pin_memory=_pin).to(
+                self.device, non_blocking=True
+            )
+        )
```
[the remark this answers is not in the document]
Comment (guzekai01) [reply python/sglang/srt/managers/schedule_batch.py:1545]: Done
